### PR TITLE
feat(ColorPicker): add destroyTooltipOnHide

### PR DESCRIPTION
**Pull request by linxianxi**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

页面中 ColorPicker 比较多时，打开后 dom 数量比较多

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     add destroyTooltipOnHide      |
| 🇨🇳 Chinese |     添加 destroyTooltipOnHide      |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 82bace3</samp>

Add a new prop `destroyTooltipOnHide` to the `ColorPicker` component, which controls whether the color state is preserved or reset when the popover is hidden. Update the English and Chinese documentation for the component accordingly.

### <samp>🤖 Generated by Copilot at 82bace3</samp>

*  Add `destroyTooltipOnHide` prop to `ColorPicker` component to control whether the popover should be destroyed when hidden or not ([link](https://github.com/ant-design/ant-design/pull/42645/files?diff=unified&w=0#diff-be92dd92152a780003d48ddc95ec6716367d327f9f890f1221b84614aaa8168fR49), [link](https://github.com/ant-design/ant-design/pull/42645/files?diff=unified&w=0#diff-be92dd92152a780003d48ddc95ec6716367d327f9f890f1221b84614aaa8168fR78), [link](https://github.com/ant-design/ant-design/pull/42645/files?diff=unified&w=0#diff-be92dd92152a780003d48ddc95ec6716367d327f9f890f1221b84614aaa8168fR132), [link](https://github.com/ant-design/ant-design/pull/42645/files?diff=unified&w=0#diff-819859f3d30fc8ec12a8ef1b55ac1d75b4b1d600eb1cee989efd5af57255b560R52), [link](https://github.com/ant-design/ant-design/pull/42645/files?diff=unified&w=0#diff-9d1ff106a032e484c8f982279c4e1295d23f69904a26e020e99310c446e93249R53))
  * Extend `ColorPickerProps` interface with `destroyTooltipOnHide` prop inherited from `PopoverProps` interface in `components/color-picker/ColorPicker.tsx` ([link](https://github.com/ant-design/ant-design/pull/42645/files?diff=unified&w=0#diff-be92dd92152a780003d48ddc95ec6716367d327f9f890f1221b84614aaa8168fR49))
  * Pass `destroyTooltipOnHide` prop to `useColorPicker` hook and use it to reset color state when popover is hidden in `components/color-picker/ColorPicker.tsx` ([link](https://github.com/ant-design/ant-design/pull/42645/files?diff=unified&w=0#diff-be92dd92152a780003d48ddc95ec6716367d327f9f890f1221b84614aaa8168fR78))
  * Pass `destroyTooltipOnHide` prop to `Popover` component and forward it to `rc-trigger` component in `components/color-picker/ColorPicker.tsx` ([link](https://github.com/ant-design/ant-design/pull/42645/files?diff=unified&w=0#diff-be92dd92152a780003d48ddc95ec6716367d327f9f890f1221b84614aaa8168fR132))
  * Document `destroyTooltipOnHide` prop and its default value in English and Chinese documentation files in `components/color-picker/index.en-US.md` and `components/color-picker/index.zh-CN.md` ([link](https://github.com/ant-design/ant-design/pull/42645/files?diff=unified&w=0#diff-819859f3d30fc8ec12a8ef1b55ac1d75b4b1d600eb1cee989efd5af57255b560R52), [link](https://github.com/ant-design/ant-design/pull/42645/files?diff=unified&w=0#diff-9d1ff106a032e484c8f982279c4e1295d23f69904a26e020e99310c446e93249R53))

### CODE REVIEW

**Comment by linxianxi** [comment]
看下是否需要改成名字，现在是直接拿的 popover 的属性名 destroyTooltipOnHide

**Comment by RedJue** [comment]
不确定需不需要，内部讨论下。

File: components/color-picker/ColorPicker.tsx [hunk 1]
```diff
@@ -46,6 +46,7 @@ export interface ColorPickerProps
   onChange?: (value: Color, hex: string) => void;
   getPopupContainer?: PopoverProps['getPopupContainer'];
   autoAdjustOverflow?: PopoverProps['autoAdjustOverflow'];
+  destroyTooltipOnHide?: PopoverProps['destroyTooltipOnHide'];
```
Comment (MadCcc) [inline components/color-picker/ColorPicker.tsx:49]: 改成 `& Pick<PopoverProps, ''>`

File: components/color-picker/ColorPicker.tsx [hunk 2]
```diff
@@ -74,6 +75,7 @@ const ColorPicker: CompoundedComponent = (props) => {
     onOpenChange,
     getPopupContainer,
     autoAdjustOverflow = true,
+    destroyTooltipOnHide,
```
Comment (MadCcc) [inline components/color-picker/ColorPicker.tsx:78]: `...popoverProps` 然后给 Popover 吧，不用一条一条写了

File: components/color-picker/ColorPicker.tsx (hunk 1 above)
Comment (linxianxi) [reply components/color-picker/ColorPicker.tsx:49]: interface 不能用 & 了呀，不然要改成 type ColorPickerProps = ...

File: components/color-picker/ColorPicker.tsx (hunk 2 above)
Comment (linxianxi) [reply components/color-picker/ColorPicker.tsx:78]: 下面已经有 popoverProps 了，而且还有 open, trigger, placement, arrow, rootClassName 这些，这里感觉还是保持这样比较好

File: components/color-picker/ColorPicker.tsx (hunk 1 above)
Comment (MadCcc) [reply components/color-picker/ColorPicker.tsx:49]: 可以的

File: components/color-picker/ColorPicker.tsx (hunk 2 above)
Comment (linxianxi) [reply components/color-picker/ColorPicker.tsx:79]: 这个是否改成 destoryPopupOnHide

**Review by MadCcc — APPROVED** [review]

File: components/color-picker/index.zh-CN.md [hunk 3]
```diff
@@ -48,6 +48,7 @@ group:
 | disabled | 禁用颜色选择器 | boolean | - |
 | placement | 弹出窗口的位置 | `top` \| `topLeft` \| `topRight` \| `bottom` \| `bottomLeft` \| `bottomRight` | `bottomLeft` |
 | arrow | 配置弹出的箭头 | `boolean \| { pointAtCenter: boolean }` | `true` | - |
+| destroyTooltipOnHide | 关闭后是否销毁弹窗 | `boolean` | `false` |
```
Comment (li-jia-nan) [inline components/color-picker/index.zh-CN.md:51]: 是不是没加版本号

**Pull request merged**